evalml/pipelines/components/transformers/transformer.py

```diff
@@ -47,7 +51,11 @@ def fit_transform(self, X, y=None):
             pd.DataFrame: Transformed X
         """
         try:
-            X_t = self._component_obj.fit_transform(X, y)
+            X2 = _convert_to_woodwork_structure(X)
```

Comment:
Allows transformers to take in all datatypes for `fit_transform`

Reply:
Ooh another we would've handled in #1406 (@angela97lin ). Thanks @bchen1116 !